### ISSUE TITLE
freeze due to name collision in module loading on case-insensitive filesystems

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaLang/julia#9007]
Very weird bug I found and I couldn't work out a minimal version to reproduce it.
It  happens on windows, but not linux : 

``` Julia
Julia Version 0.3.2+2
Commit 6babc84* (2014-10-22 01:21 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

and

``` Julia
Julia Version 0.4.0-dev+1603
Commit b923a43* (2014-11-14 11:57 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

Its about GLPlot and runtest.jl which I could reduce to only one line, namely `using GLPlot`. 
I'm using GLPlot, GLAbstraction, ModernGL, GLText, GLWindow and Reactive on master.
When I run runtest.jl either via Pkg.test("GLPlot"), or directly with julia runtests.jl, it hangs in ~95% of the cases at:
`Images/src/libmagicwand.jl/` line 11 `import Images: (@Dict)`
or in ~5% of the cases at:
`ModernGL/src/glconstants.jl` line 1 `@GenEnums begin`
If I put the file `runtest.jl` anywhere but the `test` folder, it works in 99.9% of the cases. I don't know where it stops in 0.1% of the cases, since it didn't hang anywhere else since I started investigating.
This behavior is known to me since approximately Mai, but it hardly happened, so I assumed it's a bug from my side...

**Comment by jakebolewski** [comment]
Closed by #13542

**Comment by stevengj** [comment]
@jakebolewski, I don't think this is fully resolved.   The issue above was about `include`, and `include` was not affected by #13542.  

If we want to fully address this, we need a followup PR that uses `isfile_casesensitive` in `include_from_node1`.

**Comment by stevengj** [comment]
Maybe this can be closed?

My feeling is that we should be case-sensitive (i.e. ignore the filesystem semantics) in things like `using Foo` and `Pkg.add("Foo")` where `Foo` is a semantic Julia object and the file is an implementation detail.  That's addressed by #13452 and #18210.

On the other hand, in a function like `open` or `include`, where you actually pass a pathname, my inclination is to respect the filesystem semantics (including case insensitivity).  People on case-insensitive filesystems need to know that filenames are insensitive.

The original problem, with an `import` statement being confused by a local file, was fixed in #12695 now that `import` only looks in the global path.

**Comment by SimonDanisch** [comment]
Fine by me ;) My concern from this issue _is_ fixed after all!
